tests/unit/test_ask_tell_optimization.py

```diff
@@ -171,7 +180,7 @@ def test_ask_tell_optimizer_copies_state(
     ask_tell.tell(new_data)
     state_end: Record[None] = ask_tell.to_record(copy=copy)
 
-    assert_datasets_allclose(state_start.dataset, init_dataset if copy else init_dataset + new_data)
+    assert_datasets_allclose(state_start.dataset, init_dataset)
```

Comment:
Note: this is needed due to the following change in `ask_tell_optimization.py`. The previous version updated the dataset in place, which is not the case anymore.

```
-        for tag in self._datasets:
-            self._datasets[tag] += new_data[tag]
+        self._datasets = self._acquisition_rule.update_datasets(self._datasets, new_data)
```

Reply:
(see comment about that change; it might be nice to avoid copying the dataset when we don't need to)

Reply:
Change reverted due to suggested change.